# **Financial Impact Analysis**

In [2]:
import pandas as pd

df = pd.read_csv('financial_impact.csv')

**Variables**

In [10]:
direct_losses = df['direct_loss_usd']
direct_loss_method = df['direct_loss_method']
recovery_cost = df['recovery_cost_usd']
total_loss = df['total_loss_usd']
inflation_adjusted = df['inflation_adjusted_usd']
insurance_payout = df['insurance_payout_usd']
regulatory_fine = df['regulatory_fine_usd']
total_loss_method = df['total_loss_method']
ransom_demanded = df['ransom_demanded_usd']
ransom_paid = df['ransom_paid_usd']
ransom_source = df['ransom_source']

### **Descriptive Analysis**

---

**Dataset Information**

In [8]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 778 entries, 0 to 777
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   incident_id             778 non-null    str    
 1   direct_loss_usd         778 non-null    float64
 2   direct_loss_method      778 non-null    str    
 3   ransom_demanded_usd     206 non-null    float64
 4   ransom_paid_usd         86 non-null     float64
 5   ransom_source           86 non-null     str    
 6   recovery_cost_usd       778 non-null    float64
 7   legal_fees_usd          778 non-null    float64
 8   regulatory_fine_usd     132 non-null    float64
 9   insurance_payout_usd    435 non-null    float64
 10  total_loss_usd          778 non-null    float64
 11  total_loss_method       778 non-null    str    
 12  total_loss_lower_bound  778 non-null    float64
 13  total_loss_upper_bound  778 non-null    float64
 14  inflation_adjusted_usd  778 non-null    float64
 15  

- The information regarding the contents of the dataset.

---

In [6]:
# Direct Losses
print(direct_losses.describe())

count    7.780000e+02
mean     3.652294e+07
std      1.196191e+08
min      9.000000e+04
25%      2.834239e+06
50%      8.525071e+06
75%      2.609037e+07
max      2.302300e+09
Name: direct_loss_usd, dtype: float64


- Direct losses are right-skewed
- Median = ~$8,525,071.00; 
- Losses ranged from ~$90,000.00 to ~$2,302,300,000.00 (~$2,302,210,000 difference)
---

In [7]:
# Recovery Costs
print(recovery_cost.describe())

count    7.780000e+02
mean     2.618556e+07
std      7.816093e+07
min      5.703142e+04
25%      2.055169e+06
50%      6.129803e+06
75%      1.840980e+07
max      1.238193e+09
Name: recovery_cost_usd, dtype: float64


- Recovery costs are right-skewed
- Median = ~$6,129,803.00
- Range = ~$57,031.42 to ~$1,238,193,000.00

---

In [28]:
# Total Loss and adjusted costs by inflation
print(total_loss.agg(["count", "median", "min", "max", "mean"]))
print("---------")
print(inflation_adjusted.agg(["count", "median", "min", "max", "mean"]))

count     7.780000e+02
median    1.656491e+07
min       1.737931e+05
max       3.451548e+09
mean      7.099600e+07
Name: total_loss_usd, dtype: float64
---------
count     7.780000e+02
median    1.790865e+07
min       1.737931e+05
max       3.451548e+09
mean      7.501910e+07
Name: inflation_adjusted_usd, dtype: float64


- **Total Losses**
    - Right-skewed distribution
    - Median = ~$16,564,910.00
    - Range = ~$173,793.10 - ~$3,451,548,000.00
- **Adjusted Inflation**
    - Right-skewed distribution
    - Median = ~$17,908,650.00
    - Range = ~$173,793.10 - ~$3,451,548,000.00
---

In [9]:
# Insurance Payout
print(insurance_payout.describe())


count    4.350000e+02
mean     2.665044e+07
std      7.593903e+07
min      6.280104e+04
25%      2.277000e+06
50%      6.316200e+06
75%      1.844803e+07
max      1.054416e+09
Name: insurance_payout_usd, dtype: float64


---

In [35]:
# Ransom Demanded + Paid
print(ransom_demanded.agg(["count", "median", "min", "max", "mean"]))
print("-----")
print(ransom_paid.agg(["count", "median", "min", "max", "mean"]))

count     2.060000e+02
median    1.979266e+06
min       5.000000e+04
max       7.500000e+07
mean      8.017672e+06
Name: ransom_demanded_usd, dtype: float64
-----
count     8.600000e+01
median    1.129759e+06
min       1.610098e+04
max       5.188418e+07
mean      4.497464e+06
Name: ransom_paid_usd, dtype: float64


- 206 documented ransom demands
    - Median = ~$1,979,266.00
    - Range = ~$50,000.00 - ~$75,000,000.00
- 86 documented paid ransoms
    - Median = ~$1,129,759.00
    - Range = ~$16,100.98 - ~$51,884,180.00

In [34]:
# Ransom Source
print(ransom_source.value_counts())

ransom_source
congressional testimony    19
blockchain analysis        19
FBI disclosure             15
company statement          13
insurance claim record     11
court filing                9
Name: count, dtype: int64


---

In [9]:
# Method for calculating total losses

print(total_loss_method.value_counts())

total_loss_method
estimated     369
disclosed     215
calculated    194
Name: count, dtype: int64


- The majority of total loss is estimated

---

In [12]:
print("\nMissing Values")
print("--------------------")

missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(2)

missing_df = pd.DataFrame({
    "Missing Count": missing,
    "Missing %": missing_pct
})

print(missing_df.sort_values("Missing %", ascending=False))


Missing Values
--------------------
                        Missing Count  Missing %
ransom_paid_usd                   692      88.95
ransom_source                     692      88.95
regulatory_fine_usd               646      83.03
ransom_demanded_usd               572      73.52
notes                             530      68.12
insurance_payout_usd              343      44.09
total_loss_lower_bound              0       0.00
created_at                          0       0.00
cpi_index_used                      0       0.00
inflation_adjusted_usd              0       0.00
total_loss_upper_bound              0       0.00
incident_id                         0       0.00
total_loss_method                   0       0.00
total_loss_usd                      0       0.00
direct_loss_usd                     0       0.00
legal_fees_usd                      0       0.00
recovery_cost_usd                   0       0.00
direct_loss_method                  0       0.00
updated_at                      

- The data describes all missing values in the dataset

---

**Relational Analysis**

In [ ]:
# 